ocaml/xapi/repository.ml

```diff
@@ -143,6 +161,16 @@ let sync ~__context ~self ~token ~token_id =
             Uri.make ~scheme:"file" ~path:!Xapi_globs.bundle_repository_dir ()
           in
           (Uri.to_string uri, None)
+      | `remote_pool ->
+          (* TODO: sync with Stunnel.with_client_proxy as otherwise yum
+             reposync will fail when checking the self signed certificate on
+             the remote pool. *)
+          let binary_url =
+            Printf.sprintf "http://127.0.0.1:%s/repository/%s/"
```

Comment:
Use the URI module to construct the URL. Is this compatible with IPv6?

Reply:
This piece of code was copied from `repository_helpers.with_local_repositories`, and I think "XenServer doesn’t support the use of IPv6 in its Control Domain (dom0)", so "127.0.0.1" will always work.
But as it is not related to this PR, remove this piece of code and leave it to later PRs.

Reply:
We really want to avoid the manual construction of URLs -e even when trivial like this one.

Reply:
The code is hosted in xapi-project, not xenserver. There are other downstream projects that do use IPv6. Please be mindful of this and change the code accordingly.